### ISSUE TITLE
can't reproduce the performance and time (about EfficientLoFTR)

### ISSUE THREAD

**Issue by krbangzi** [leoluxxx/JamMa#14]
Hi, thank you for your open source,  I'm following your work!

I follow your environment installation and special experimental setup (image size 832x832, LO-RANSAC with 0.5 px threshold), but I can't reproduce the performance and time on the MegaDepth dataset reported in your paper. The difference is that I use a 3090 GPU.

I got results as follow:

**JamMa**
run your script got:
**{'auc@10': 0.7736561156000842,
 'auc@20': 0.865181405819172,
 'auc@5': 0.6419782203313796,
 'num_matches@1e-04': 4021.424666666667,
 'prec@1e-04': 0.8891372942176896}**
Averaged Matching time over 1500 pairs: **85.92 ms**

**Efficient LoFTR**
I modify their scripts outdoor_full_auc.sh to follow your experimental setup (size=832,  lo-ransac, ransac_times=1, RANSAC_PIXEL_THR=0.5)
and got:
**{'auc@10': 0.7727105929415025,
 'auc@20': 0.8657305035091719,
 'auc@5': 0.6412782690733704,
 'num_matches': 1757.206,
 'prec@5e-04': 0.9671603955361585}**
Averaged Matching time over 1500 pairs: **61.47 ms**

But in your paper Tab.2 (on 4090 GPU),
auc@5 / auc@10 / auc@20 /     time
**JamMa: 64.1 / 77.4 / 86.5    /  59.9ms
EfficientLoFTR: 63.7 / 77.0 / 86.4   /  69.6ms**

I think it is normal for them to be slower on 3090 than on 4090, but it is unusual EfficientLoFTR got better results and faster than JamMa in the same setting.
The main problem is that the results of efficientloftr are inconsistent with those in your paper. Even EfficientLoFTR run 61.47ms on 3090 and  69.6ms on 4090.

Is there anything I did wrong or missed?  And did anyone successfully reproduce the results?
Looking forward to your reply, thanks a lot!

**Comment by leoluxxx** [comment]
Thank you for following our work!
It looks like results of JamMa are normal.
Please check a few configurations of ELoFTR: we 1. don't enable any mixed precision (enabled in eloftr_full.py). 2. don't use flash attention. 3. don't skip Softmax. 4. don't do warmups.
Looking forward to your reply.

**Comment by leoluxxx** [comment]
Here are my reproduction results on the 3090 GPU (Pytorch 2.0.0, CUDA 11.8) for your reference.

ELoFTR without mp:
Averaged Matching time over 1500 pairs: 83.11 ms
{'auc@10': 0.771880778654406,
 'auc@20': 0.8644004095669467,
 'auc@5': 0.6395993376020849,
 'num_matches': 1634.8893333333333,
 'prec@5e-04': 0.9802894429573493}

Averaged Matching time over 1500 pairs: 82.78 ms
2025-05-19 21:22:28.183 | INFO     | src.utils.metrics:aggregate_metrics:247 - Aggregating metrics over 1500 unique items...
2025-05-19 21:22:28.204 | INFO     | src.lightning.lightning_loftr:test_epoch_end:272 - 
{'auc@10': 0.7742462577468341,
 'auc@20': 0.8673848796927459,
 'auc@5': 0.641740296951199,
 'num_matches': 1634.8893333333333,
 'prec@1e-04': 0.9667559099837478}
ELoFTR with mp:
Averaged Matching time over 1500 pairs: 68.16 ms
{'auc@10': 0.7724043567623361,
 'auc@20': 0.866123684511599,
 'auc@5': 0.6355081723990822,
 'num_matches': 1635.0113333333334,
 'prec@1e-04': 0.9667623017536341}

JamMa without mp:
Averaged Matching time over 1500 pairs: 85.18 ms
{'auc@10': 0.7735685615107097,
 'auc@20': 0.8651048398358083,
 'auc@5': 0.6423284512306672,
 'num_matches@1e-04': 4021.423333333333,
 'prec@1e-04': 0.8891323362369127}

JamMa with mp:
Averaged Matching time over 1500 pairs: 61.30 ms
{'auc@10': 0.773438541676584,
 'auc@20': 0.8646834118255085,
 'auc@5': 0.641765416386611,
 'num_matches@1e-04': 4049.462,
 'prec@1e-04': 0.8890652040358009}

**Comment by leoluxxx** [comment]
Here are my reproduction results on the 3090 GPU (Pytorch 2.0.0, CUDA 11.8) for your reference.

ELoFTR without mp:
Averaged Matching time over 1500 pairs: 83.11 ms
{'auc@10': 0.771880778654406,
'auc@20': 0.8644004095669467,
'auc@5': 0.6395993376020849,
'num_matches': 1634.8893333333333,
'prec@1e-04': 0.9667559099837478}

JamMa without mp:
Averaged Matching time over 1500 pairs: 85.18 ms
{'auc@10': 0.7735685615107097,
'auc@20': 0.8651048398358083,
'auc@5': 0.6423284512306672,
'num_matches@1e-04': 4021.423333333333,
'prec@1e-04': 0.8891323362369127}

ELoFTR with mp:
Averaged Matching time over 1500 pairs: 68.16 ms
{'auc@10': 0.7724043567623361,
'auc@20': 0.866123684511599,
'auc@5': 0.6355081723990822,
'num_matches': 1635.0113333333334,
'prec@1e-04': 0.9667623017536341}

JamMa with mp:
Averaged Matching time over 1500 pairs: 61.30 ms
{'auc@10': 0.773438541676584,
'auc@20': 0.8646834118255085,
'auc@5': 0.641765416386611,
'num_matches@1e-04': 4049.462,
'prec@1e-04': 0.8890652040358009}

**Comment by leoluxxx** [comment]
I used a unified coarse threshold of 0.2 for all semi-dense methods, which should be the reason :)

**Comment by krbangzi** [comment]
Yes, I reproduced this result. 
I got num_matches=1634 by set thr=0.2.

Thanks for your help.
